File: codegen/codegen-erlang/src/test/resources/ir/handler_discovery_init_handlers.expected.erl
-spec init_handlers() -> ok | {error, term()}.
init_handlers() ->
    case resolve_impl(?DEFAULT_IMPL) of
        {ok, Handlers} ->
            persistent_term:put(?HANDLERS_KEY, Handlers),
            ok;
        {error, Reason} ->
            persistent_term:put(?HANDLERS_KEY, #{}),
            {error, Reason}
    end.

File: codegen/codegen-elixir/src/test/resources/ir/handler_discovery_init_handlers.expected.ex
@spec init_handlers() :: :ok | {:error, term()}
def init_handlers do
  case resolve_impl(@default_impl) do
    {:ok, handlers} ->
      :persistent_term.put(@handlers_key, handlers)
      :ok
    {:error, reason} ->
      :persistent_term.put(@handlers_key, %{})
      {:error, reason}
  end
end
